File: app/src/main/java/com/github/nikololoshka/pepegaschedule/modulejournal/view/data/MarkType.java
package com.github.nikololoshka.pepegaschedule.modulejournal.view.data;

import androidx.annotation.NonNull;

/**
 * Типы оценок в модульном журнале.
 */
public enum MarkType {

    /**
     * Первый модуль.
     */
    FIRST_MODULE,

    /**
     * Второй модуль.
     */
    SECOND_MODULE,

    /**
     * Курсовая работа.
     */
    COURSEWORK,

    /**
     * Зачёт.
     */
    CREDIT,

    /**
     * Экзамен.
     */
    EXAM;

    /**
     * Возвращает тип оценки из перечисления соответстующего значению ответа от сервера.
     * @param value ответ от сервера.
     * @return тип оценки.
     */
    public static MarkType of(@NonNull String value) {
        switch (value) {
            case "М1":
                return FIRST_MODULE;
            case "М2":
                return SECOND_MODULE;
            case "К":
                return COURSEWORK;
            case "З":
                return CREDIT;
            case "Э":
                return EXAM;
        }
        throw new IllegalArgumentException("Unknown mark type: " + value);
    }
}

File: domain/journal-core/src/main/java/com/vereshchagin/nikolay/stankinschedule/journal/core/domain/model/MarkType.kt
package com.vereshchagin.nikolay.stankinschedule.journal.core.domain.model

/**
 * Типы оценок в модульном журнале.
 */
enum class MarkType(val tag: String, val weight: Int) {

    /**
     * Первый модуль.
     */
    FIRST_MODULE("М1", 3),

    /**
     * Второй модуль.
     */
    SECOND_MODULE("М2", 2),

    /**
     * Курсовая работа.
     */
    COURSEWORK("К", 5),

    /**
     * Зачёт.
     */
    CREDIT("З", 5),

    /**
     * Экзамен.
     */
    EXAM("Э", 7);

    override fun toString(): String {
        return tag
    }

    companion object {
        /**
         * Возвращает тип оценки из перечисления соответствующего значению ответа от сервера.
         */
        @JvmStatic
        fun of(value: String): MarkType {
            for (type in values()) {
                if (type.tag == value) {
                    return type
                }
            }

            if (value != null && value.isEmpty()) {
                return EXAM
            }

            throw IllegalArgumentException("Unknown mark type: '$value'")
        }
    }
}